jax/interpreters/pxla.py

```diff
@@ -130,39 +199,23 @@ def _shard_abstract_array(size, x):
   return ShapedArray(x.shape[1:], x.dtype)
 shard_aval_handlers[ShapedArray] = _shard_abstract_array
 
-def aval_to_result_handler(size, nrep, aval):
+def aval_to_result_handler(size, sharding_spec, indices, aval):
   try:
-    return pxla_result_handlers[type(aval)](size, nrep, aval)
+    return pxla_result_handlers[type(aval)](size, sharding_spec, indices, aval)
   except KeyError as err:
     raise TypeError("No pxla_result_handler for type: {}".format(type(aval))
                     ) from err
 PxlaResultHandler = Callable[..., Callable[[Any], Any]]
```

Comment:
This would be an effective place to put a comment, because it sits in between all the definitions and all the uses of result handlers.

Reply:
I documented `aval_to_result_handler`, and updated this typedef to be more specific. Lemme know if you think more comments would help.